### PR TITLE
Add tests and Unicode support

### PR DESCRIPTION
**Pull request by lyrahgames**
- Use build2 version 0.16.0 and update the manifest
- Add upstream unit tests to the build system
- Add configuration for Unicode support
- Update usage in README.md

[Here](https://ci.cppget.org/@bfdd482b-a5dd-4928-ae7b-311093adaa79), you can find the CI build of this commit.

I hope the source code changes are ok for you.
If so, either you or me can release a new version on cppget.org.
Thank you so much!

### CODE REVIEW

File: libcxxopts/include/buildfile [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
 }
+
+# Set macro for Unicode configuration in the header.
+#
+if $config.libcxxopts.use_unicode
+  lib{cxxopts}: cxx.export.poptions += -DCXXOPTS_USE_UNICODE
+
+# Installation is correctly handled by default.
+#
+# hxx{cxxopts}@./: install = include/
```
Comment (Rookfighter) [inline libcxxopts/include/buildfile]: Remove unused outcommented code.

**Review by Rookfighter — CHANGES_REQUESTED** [review]
Looks good to me :+1:  Could you also fix the GitHub CI builds?

I just checked the logs and realized that `cxx.std = 17` is defined in the msvc configs [here](https://github.com/build2-packaging/cxxopts/blob/master/build-config). Removing the line in both configs should fix the msvc build.

The MacOS clang builds use `c++1z` (afaik = c++17), which is set [here](https://github.com/build2-packaging/cxxopts/blob/master/.github/workflows/build2.yml#L69). Not sure what my past-me was doing and why all this is so messy. I guess I was trying things out with the auto_ptr issue and those are some left-overs.

**Comment by lyrahgames** [comment]
> Looks good to me +1 

Thank you ^^

> Could you also fix the GitHub CI builds?

I totally understand the idea and the urge of experimenting with a build2 setup for GitHub Actions.
But to be honest, as build2 changes, the GitHub CI builds are likely to break again.
This will only increase time spent on this repository in the future.
In my opinion, the build2 CI is is the best reference for build2 packages because it is developed alongside build2 itself.
Presumably, it would be more appropriate to use GitHub Actions in private repositories that exhibit build2 packages.
What do you think?

> I just checked the logs and realized that `cxx.std = 17` is defined in the msvc configs [here](https://github.com/build2-packaging/cxxopts/blob/master/build-config). Removing the line in both configs should fix the msvc build.
> 
> The MacOS clang builds use `c++1z` (afaik = c++17), which is set [here](https://github.com/build2-packaging/cxxopts/blob/master/.github/workflows/build2.yml#L69). Not sure what my past-me was doing and why all this is so messy. I guess I was trying things out with the auto_ptr issue and those are some left-overs.

I removed the lines from the config as it has not been a big issue. Hopefully, it does work now.
For the case it doesn't, I would like to step back from fixing this problem, as I am not familiar with GitHub Actions.

**Review by Rookfighter — APPROVED** [review]

**Comment by Rookfighter** [comment]
> What do you think?

I get your point, though the nice integration with GitHub PRs is the reason why some contributors enjoy GitHub Actions (like me). E. g. just note that you did not post another bdep ci link after your latest commits on this PR. Usually the right way would have been to request another link before merging. Though the new changes were minor, GitHub CI was happy and you seem way more experienced with build2 than I am, so I didn't see a problem there ;) 
IMO automation in CI and PRs is key to make everybody's life easier. The current build2 integration into GitHub Actions is not optimal, I think. A better (but not yet available) solution would be to have GitHub Workflows trigger bdep ci and report the results. Se a lengthy discussion [here](https://github.com/build2/setup-build2-github-action/issues/1). 
Still I like having a simple GitHub Workflow as smoke test which gives me direct visual feedback if a PR is not fundamentally broken.
And I wouldn't like to merge PRs with broken CI builds :D so either fix it or disable it (with working bdep ci results) would be my preference

**Comment by boris-kolpackov** [comment]
> A better (but not yet available) solution would be to have GitHub Workflows trigger bdep ci and report the results.

Strongly agree. And we are willing to actively work on that from the `build2` side if someone can "meet" us on the GH Actions side. If you would be willing to help, let me know.


> And I wouldn't like to merge PRs with broken CI builds :D so either fix it or disable it (with working bdep ci results) would be my preference

Looks like we might need some policy regarding this for packages hosted under `build2-packaging/`. I can see the validity of both points of view here: On one hand, automatically checking PRs is a reasonable thing to expect these days and until we have proper integration (per the above), GH Actions is probably the most viable option. On the other hand, `build2` CI is the "reference" CI for `build2` packages and having to take care of additional CI integrations that one may not even be knowledgeable of, places an unnecessary burden on contributors. So I am leaning towards the "it's reasonable to ask but not to request" fixes for other CI setups that the project may use (and of course, it's reasonable to request a `build2` CI URL for a PR). 

Admittedly this causes a bit of friction but hopefully it's temporary until we have a proper integration of GH Actions with `build2` CI. At which point it will hopefully become less likely that someone would want to use yet another CI stetup (but you never know).

What do you think?

**Comment by lyrahgames** [comment]
> I get your point, though the nice integration with GitHub PRs is the reason why some contributors enjoy GitHub Actions (like me). E. g. just note that you did not post another bdep ci link after your latest commits on this PR. Usually the right way would have been to request another link before merging. Though the new changes were minor, GitHub CI was happy and you seem way more experienced with build2 than I am, so I didn't see a problem there ;) IMO automation in CI and PRs is key to make everybody's life easier.

First, thanks for your thorough answer and the merge!
I really appreciate this discussion and I do like the idea of CI automation very much.
Because of that, two years ago, I experimented myself with CircleCI-based CI automation for build2 packages in a [private project](https://github.com/lyrahgames/pxart/blob/master/.circleci/config.yml).
For that, I built Docker images for [Clang](https://github.com/lyrahgames/docker-ubuntu-clang-build2) and [GCC](https://github.com/lyrahgames/docker-ubuntu-gcc-build2) on Ubuntu.
At that time, I also set up an automated build step on Docker Hub.
As soon as GitHub Actions were introduced, I also wanted to take a look into it but did not really have the time to do.

> The current build2 integration into GitHub Actions is not optimal, I think. A better (but not yet available) solution would be to have GitHub Workflows trigger bdep ci and report the results.

Indeed, I would love that feature.

> Se a lengthy discussion [here](https://github.com/build2/setup-build2-github-action/issues/1).

There is more happening then I was thinking at first. I will take my time to read through it again.
Maybe I can also try to help.

> > A better (but not yet available) solution would be to have GitHub Workflows trigger bdep ci and report the results.
> 
> Strongly agree. And we are willing to actively work on that from the `build2` side if someone can "meet" us on the GH Actions side. If you would be willing to help, let me know.

I am willing to help but there is no real experience with GitHub Actions on my side.

> > And I wouldn't like to merge PRs with broken CI builds :D so either fix it or disable it (with working bdep ci results) would be my preference
> 
> Looks like we might need some policy regarding this for packages hosted under `build2-packaging/`. I can see the validity of both points of view here: On one hand, automatically checking PRs is a reasonable thing to expect these days and until we have proper integration (per the above), GH Actions is probably the most viable option. On the other hand, `build2` CI is the "reference" CI for `build2` packages and having to take care of additional CI integrations that one may not even be knowledgeable of, places an unnecessary burden on contributors. So I am leaning towards the "it's reasonable to ask but not to request" fixes for other CI setups that the project may use (and of course, it's reasonable to request a `build2` CI URL for a PR).

I would like such a policy.
We could create Markdown files with some contribution guidelines for the build2-packaging organization in the repository `build2-packaging/.github` to always provide these as default files in other repositories.

> Admittedly this causes a bit of friction but hopefully it's temporary until we have a proper integration of GH Actions with `build2` CI.

As long as everyone communicates in a healthy manner with each other, there shouldn't be too much friction, I think ^^
